db/dcrsqlite/dcrsqlite_test.go

```diff
@@ -30,3 +32,42 @@ func InitTestDB(targetDBFile string) *DB {
 	}
 	return db //is not nil
 }
+
+// checkChartsDataIsDefault checks if chartsData is in default mode:
+//  - all the returned arrays expected to be empty
+//  - chartsData is not null (nil)
+//
+// These checks are sorted according to the ChartsData struct fields order
+func checkChartsDataIsDefault(testedFunctionName string, chartsData *dbtypes.ChartsData) {
+	if chartsData == nil {
+		testutil.ReportTestFailed(
+			testedFunctionName + "() failed: value is nil")
+	}
+	// Array checks:
+	ExpectEmptyArray(testedFunctionName, "chartsData.TimeStr", chartsData.TimeStr)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Difficulty", chartsData.Difficulty)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Time", chartsData.Time)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Value", chartsData.Value)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Size", chartsData.Size)
+	ExpectEmptyArray(testedFunctionName, "chartsData.ChainSize", chartsData.ChainSize)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Count", chartsData.Count)
+	ExpectEmptyArray(testedFunctionName, "chartsData.SizeF", chartsData.SizeF)
+	ExpectEmptyArray(testedFunctionName, "chartsData.ValueF", chartsData.ValueF)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Unspent", chartsData.Unspent)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Revoked", chartsData.Revoked)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Height", chartsData.Height)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Pooled", chartsData.Pooled)
+	ExpectEmptyArray(testedFunctionName, "chartsData.Solo", chartsData.Solo)
+}
+
+func ExpectEmptyArray(
```

Comment:
Why does this have to take an interface and use reflect / .Len when you could simply test with `len`?  e.g. 

```if len(chartsData.TimeStr) != 0```

Reply:
All the arrays above has different types: `uint[]`, `float[]`, `string[]` and so on. Reflect helps here to avoid creating a dedicated function for each array type.

Reply:
As I mentioned in my other comment, you don't need a dedicated function at all. `len(uintArray)`, `len(floatArray)`, all fine.